instance/models/server.py

```diff
@@ -316,18 +319,23 @@ def update_status(self):
 
         # First check if it makes sense to update the current status.
         # This is not the case if we can not interact with the server:
-        if self.status == Status.BuildFailed:
+        if self.status in (Status.BuildFailed, Status.Terminated):
+            return self.status
+        try:
+            os_server = self.os_server
+        except (requests.RequestException, novaclient.exceptions.ClientException):
+            self.logger.debug('Could not reach the OpenStack API')
```

Comment:
When I disconnected my IM server from the internet, I could see my app server status go from `ready` to `unknown` in the database, but the GUI never showed `Unknown`.  It just got stuck at "Loading appserver details..."

The log showed the retries, and the debug message above though:

```
17/Sep/2016 10:19:41 |  WARNING | process=28842 | requests.packages.urllib3 | Retrying (Retry(total=9, connect=9, read=10, redirect=10)) after connection broken by 'NewConnectionError('<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7f3b361b0ac8>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /v2.0/tokens
17/Sep/2016 10:20:01 |  WARNING | process=28842 | requests.packages.urllib3 | Retrying (Retry(total=8, connect=8, read=10, redirect=10)) after connection broken by 'NewConnectionError('<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7f3b36198cf8>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /v2.0/tokens
17/Sep/2016 10:20:21 |  WARNING | process=28842 | requests.packages.urllib3 | Retrying (Retry(total=7, connect=7, read=10, redirect=10)) after connection broken by 'NewConnectionError('<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7f3b350dfcc0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /v2.0/tokens
17/Sep/2016 10:20:41 |  WARNING | process=28842 | requests.packages.urllib3 | Retrying (Retry(total=6, connect=6, read=10, redirect=10)) after connection broken by 'NewConnectionError('<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7f3b34e12198>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /v2.0/tokens
17/Sep/2016 10:21:01 |  WARNING | process=28842 | requests.packages.urllib3 | Retrying (Retry(total=5, connect=5, read=10, redirect=10)) after connection broken by 'NewConnectionError('<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7f3b34e12208>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /v2.0/tokens
17/Sep/2016 10:21:21 |  WARNING | process=28842 | requests.packages.urllib3 | Retrying (Retry(total=4, connect=4, read=10, redirect=10)) after connection broken by 'NewConnectionError('<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7f3b34e120b8>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /v2.0/tokens
17/Sep/2016 10:21:41 |  WARNING | process=28842 | requests.packages.urllib3 | Retrying (Retry(total=3, connect=3, read=10, redirect=10)) after connection broken by 'NewConnectionError('<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7f3b34e122b0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /v2.0/tokens
17/Sep/2016 10:22:01 |  WARNING | process=28842 | requests.packages.urllib3 | Retrying (Retry(total=2, connect=2, read=10, redirect=10)) after connection broken by 'NewConnectionError('<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7f3b34e121d0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /v2.0/tokens
17/Sep/2016 10:22:21 |  WARNING | process=28842 | requests.packages.urllib3 | Retrying (Retry(total=1, connect=1, read=10, redirect=10)) after connection broken by 'NewConnectionError('<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7f3b34e12470>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /v2.0/tokens
17/Sep/2016 10:22:41 |  WARNING | process=28842 | requests.packages.urllib3 | Retrying (Retry(total=0, connect=0, read=10, redirect=10)) after connection broken by 'NewConnectionError('<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7f3b34e120f0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /v2.0/tokens
17/Sep/2016 10:23:01 |    DEBUG | process=28842 | instance.models.server    | server=edxapp-preview-jille | Could not reach the OpenStack API
```